### PR TITLE
Format Code with Prettier and use Stricter Linter

### PR DESCRIPTION
**Pull request by hipstersmoothie**
* Add prettier for consistent code formatting. Make contributing and reading the code a lot easier
	* took out editor config in favor of prettier
* Switch from eslint to XO. provides a far stricter linter and conforms better to modern JS practices. 
* Add babel-preset-env to get ES features. I haven't actually modified the code to be to ESexy yet. That'll come in another PR

@edi9999 Would you like to review?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,24 +49,22 @@
...
-    "babel-preset-es2015": "^6.0.14",
-    "babel-preset-stage-0": "^6.0.14",
+    "babel": "6.23.0",
+    "babel-cli": "6.26.0",
+    "babel-core": "^6.26.3",
+    "babel-preset-es2015": "6.24.1",
+    "babel-preset-stage-0": "6.24.1",
     "browserify": "^14.3.0",
     "envify": "^3.4.0",
-    "eslint": "^3.14.1",
```
Comment (edi9999) [inline package.json]: Why eslint-config-xo-react ?

Comment (hipstersmoothie) [reply package.json]: this comment has been addressed

**Comment by edi9999** [comment]
> @edi9999 hwo you you feel about me adding the browser/lib/*.js to the gitignore. It is really weird that the built versions of those files are committed. this would fix #433. I could set it up to only do it before a publish

yes, +1 on this, I don't see why this would be commited each time.

**Comment by hipstersmoothie** [comment]
@edi9999 I'll exclude browser/lib/*.js in another PR. how does this one look now? 

* I put back the editorconfig and reformatted
* Made prettier run on commit and ignore the built files for now
* Made sure all linting and testing passes

can you approve the changes?

**Comment by edi9999** [comment]
Thanks @hipstersmoothie  !

**Pull request merged**